src/Integration.Vsix/Analysis/IssueConsumerFactory_IssueHandler.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.VisualStudio.Text;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Core.Suppression;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.Analysis
+{
+    partial class IssueConsumerFactory
+    {
+        internal class IssueHandler
+        {
+            // We can't mock the span translation code (to difficult to mock),
+            // so we use this delegate to by-passes it in tests.
+            // See https://github.com/SonarSource/sonarlint-visualstudio/issues/1522
+            internal /* for testing */ delegate IEnumerable<IAnalysisIssueVisualization> TranslateSpans(IEnumerable<IAnalysisIssueVisualization> issues, ITextSnapshot activeSnapshot);
+
+            private readonly ITextDocument textDocument;
+            private readonly string projectName;
+            private readonly Guid projectGuid;
+            private readonly IIssuesFilter issuesFilter;
+            private readonly PublishSnapshot publishSnapshot;
+
+            private readonly TranslateSpans translateSpans;
+
+            public IssueHandler(ITextDocument textDocument,
+                string projectName,
+                Guid projectGuid,
+                IIssuesFilter issuesFilter,
+                PublishSnapshot publishSnapshot)
+                : this (textDocument, projectName, projectGuid, issuesFilter, publishSnapshot, DoTranslateSpans)
+            {
+            }
+
+            internal /* for testing */ IssueHandler(ITextDocument textDocument,
+                string projectName,
+                Guid projectGuid,
```

Comment:
Change - the `projectName` and `projectGuid` are now passed to the issue handling class in the constructor, rather than being fetched each time in the `HandleNewIssues` method.